pkg/koordlet/resmanager/resctrl_reconcile.go

```diff
@@ -236,31 +249,49 @@ func (r *ResctrlReconcile) calculateAndApplyCatMbPolicyForGroup(group string, l3
 		return nil
 	}
 	// calculate updating resource
-	resource := executor.CalculateMbSchemataResource(group, memBwPercent, l3Num)
+	resource, err := resourceexecutor.NewResctrlMbSchemataResource(group, memBwPercent, l3Num)
+	if err != nil {
+		klog.V(4).Infof("failed to get mba schemata resource for group %s, err: %s", group, err)
+		return err
+	}
+	if resource == nil {
+		klog.V(5).Infof("apply mba policy for group %s skipped, schemata %v, l3 number %v, no need to update",
+			group, memBwPercent, l3Num)
+		return nil
+	}
 
 	// write policy into resctrl files if need update
-	isUpdated, err := r.executor.UpdateByCache(resource)
+	isUpdated, err := r.executor.Update(false, resource)
```

Comment:
seems llc/mba rule can be update with cache=true

Reply:
It is not cacheable anymore. L3 CAT and MBA rules can both update part of the schemata of the same file, which will generate two updates with the same cache key (`Path()`). However, now we only generate the new updater if the schemata of L3 CAT or MBA should change.

Reply:
how about not use `Path()` as the key.

Reply:
fixed. now we firstly generate a ResctrlResourceUpdater, then the updater will update the schemata if content is different